components/lib/utils/ObjectUtils.js

```diff
@@ -416,4 +416,25 @@ export default class ObjectUtils {
 
         return [];
     }
+
+    static mutateObjectProperty(prop, value, obj) {
+        // Make sure types are correct
+        if (typeof obj === 'object' && typeof prop === 'string') {
+            const _prop = prop.split('.');
+
+            // Exit if for some reason the array is empty
+            if (_prop.length < 1) return;
+
+            // If key does not exist create and empty object
+            if (!obj[_prop[0]]) {
+                obj[_prop[0]] = {};
+            }
+
+            if (_prop.length === 1) {
+                return (obj[_prop[0]] = value);
+            }
+
+            this.mutateObjectProperty(_prop.slice(1).join('.'), value, obj[_prop[0]]);
+        }
+    }
```

Comment:
@bu3alwa isn't this method similar to what `ObjectUtils.resolveFieldData` is doing?

Reply:
Similar but my function mutates the object vs this creates a new `object | null` . And because of this line 

https://github.com/primefaces/primereact/pull/4818/files#diff-5c774f14f0f649e0edd947820d6ab8bf8ec6605b70edd15a4e18feb77e422f1cL282

I haven't used `resolveFieldData` I guess. It looks like its changing the object by reference and I haven't dug deep enough to figure out why that is or if its even needed. but I assumed its correct and created that function to mutate deeply nested properties.

Let me know if you think there is a better approach.

Reply:
I have to study the problem more but it looks like resolveFieldData is exactly what they use for nested properties so `const value = ObjectUtils.resolveFieldData(json, "your.nested.prop");` would traverse the props until it gets the value?

Reply:
Oh i see this is on editing.  Maybe call it `mutateFieldData` to be consistent and it looks like you can reuse a lot of that code?

Reply:
Not sure if that works unless I'm blind. I don't see any recursion there. If it's nested `N` deep the two for loops won't go deeper than depth 2.

Reply:
in resolveFieldData it keeps looping over the split "." until it finds a value no?

```js
let fields = field.split('.');
                let value = data;

                for (var i = 0, len = fields.length; i < len; ++i) {
                    if (value == null) {
                        return null;
                    }

                    value = value[fields[i]];
                }
                ```

Reply:
Oh ok so I was blind I'll update it tonight or tomorrow then.